includes/class-migration.php

```diff
@@ -0,0 +1,101 @@
+<?php
+namespace Activitypub;
+
+class Migration {
+	public static function init() {
+		\add_action( 'activitypub_schedule_migration', array( self::class, 'maybe_migrate' ) );
+	}
+
+	public static function get_target_version() {
+		return get_plugin_version();
+	}
+
+	public static function get_version() {
+		return get_option( 'activitypub_db_version', 0 );
+	}
+
+	/**
+	 * Whether the database structure is up to date.
+	 *
+	 * @return bool
+	 */
+	public static function is_latest_version() {
+		return (bool) version_compare(
+			self::get_version(),
+			self::get_target_version(),
+			'=='
+		);
+	}
+
+	/**
+	 * Updates the database structure if necessary.
+	 */
+	public static function maybe_migrate() {
+		if ( self::is_latest_version() ) {
+			return;
+		}
+
+		$version_from_db = self::get_version();
+
+		if ( version_compare( $version_from_db, '1.0.0', '<' ) ) {
+			self::migrate_from_0_17();
+			self::migrate_from_0_16();
+		}
```

Comment:
Ideally, you can just extend a construct like this:
```suggestion
		if ( version_compare( $version_from_db, '0.16.0', '<' ) ) {
			self::migrate_from_0_16();
		}
		if ( version_compare( $version_from_db, '0.17.0', '<' ) ) {
			self::migrate_from_0_17();
		}
```